### DISCUSSION TITLE
Dealing with additionalProperties

### DISCUSSION THREAD

**Discussion by rupinder10** [go-swagger/go-swagger#2829] [Q&A]
I have swagger file with the following definition:

`
"Components": {
      "type": "object",
      "properties": {
        "schemas": {
          "description": "An object to hold reusable schema objects",
          "type": "object",
          "additionalProperties": {
            "$ref": "#/definitions/Schema"
          }
        }
`

And input to unmarshal is

 `{
"schemas": {
  "OrderHeader": {
    "type": "GatewaySchema",
    "schema": "Test1"
  },
  "OrderLine": {
    "type": "GatewaySchema",
    "schema": "Test2"
  },
  "order_POST_bodyParameters": {
    "type": "GatewaySchema",
    "schema": "Test3"
  },
  "order_POST_response": {
    "type": "GatewaySchema",
    "schema": "Test4"
  }
}
}`

The following piece of code throws an error: `json: cannot unmarshal object into Go struct field Components.schemas of type models.Schema`

`	buf, err := ioutil.ReadFile("order.json")
	reader := bytes.NewReader(buf)
	if err == nil {
		dec := json.NewDecoder(reader)
		dec.UseNumber() // preserve number formats
		obj := &models.Components{}
		err := dec.Decode(obj)
`

I have read up everything and seems like additionalProperties are supported in go-swagger code generation. Obviously, there is something I am missing regarding this.

The swagger file used to generate the code and the project with the generate code and the client code to test is attached.

[ServiceManagementBody.zip](https://github.com/go-swagger/go-swagger/files/9508353/ServiceManagementBody.zip)
[wmapis.zip](https://github.com/go-swagger/go-swagger/files/9508354/wmapis.zip)

**Comment by rupinder10** [comment]
I did see that additionalProperties are supported in go-swagger. And they seem to be in the generated code too. But for some reason they can't be parsed ? And there are no details in the error message

**Comment by casualjim** [comment]
You are not sending schemas. it appears to me that you are trying to have some form of inheritance going?

**Reply by rupinder10** [reply]
You are correct. "GatewaySchema" has "allOf" from "Schema". So it inherits from "Schema". If you see the spec, Components has a property called "schemas". The property is of type "Schema" which has a discriminator "type" which refers to the concrete type "GatewaySchema". So, the input is correct and so is that spec. And it generates valid models. But during parsing it fails to parse.   I can change the input but dont know what to change it to make it work as per the spec and the generated code.
